// ng2-bootstrap/components/modal/modal-backdrop.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/modal/modal-backdrop.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/element';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
export class Wrapper_ModalBackdropComponent {
  context:import0.ModalBackdropComponent;
  changed:boolean;
  constructor(p0:any,p1:any,p2:any) {
    this.changed = false;
    this.context = new import0.ModalBackdropComponent(p0,p1,p2);
  }
  detectChangesInInputProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    return changed;
  }
  detectChangesInHostProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
}
var renderType_ModalBackdropComponent_Host:import2.RenderComponentType = (null as any);
class _View_ModalBackdropComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import3.AppElement;
  _ModalBackdropComponent_0_4:Wrapper_ModalBackdropComponent;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_ModalBackdropComponent_Host0,renderType_ModalBackdropComponent_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.selectOrCreateRenderHostElement(this.renderer,'bs-modal-backdrop',new import4.InlineArray2(2,'class','modal-backdrop'),rootSelector,(null as any));
    this._appEl_0 = new import3.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_ModalBackdropComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._ModalBackdropComponent_0_4 = new Wrapper_ModalBackdropComponent(this.parentInjector.get(import0.ModalBackdropOptions),new import8.ElementRef(this._el_0),this.renderer);
    this._appEl_0.initComponent(this._ModalBackdropComponent_0_4.context,([] as any[]),compView_0);
    compView_0.create(this._ModalBackdropComponent_0_4.context,this.projectableNodes,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.ModalBackdropComponent) && (0 === requestNodeIndex))) { return this._ModalBackdropComponent_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._ModalBackdropComponent_0_4.detectChangesInInputProps(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this._ModalBackdropComponent_0_4.detectChangesInHostProps(this,this._el_0,throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_ModalBackdropComponent_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  if ((renderType_ModalBackdropComponent_Host === (null as any))) { (renderType_ModalBackdropComponent_Host = viewUtils.createRenderComponentType('',0,import9.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_ModalBackdropComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const ModalBackdropComponentNgFactory:import10.ComponentFactory<import0.ModalBackdropComponent> = new import10.ComponentFactory<import0.ModalBackdropComponent>('bs-modal-backdrop',viewFactory_ModalBackdropComponent_Host0,import0.ModalBackdropComponent);
const styles_ModalBackdropComponent:any[] = ([] as any[]);
var renderType_ModalBackdropComponent:import2.RenderComponentType = (null as any);
class _View_ModalBackdropComponent0 extends import1.AppView<import0.ModalBackdropComponent> {
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_ModalBackdropComponent0,renderType_ModalBackdropComponent,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this.init(([] as any[]),([] as any[]),([] as any[]),([] as any[]));
    return (null as any);
  }
}
export function viewFactory_ModalBackdropComponent0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<import0.ModalBackdropComponent> {
  if ((renderType_ModalBackdropComponent === (null as any))) { (renderType_ModalBackdropComponent = viewUtils.createRenderComponentType('',0,import9.ViewEncapsulation.None,styles_ModalBackdropComponent,{})); }
  return new _View_ModalBackdropComponent0(viewUtils,parentInjector,declarationEl);
}
