// ng2-bootstrap/components/pagination/pager.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/pagination/pager.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/element';
import * as import4 from '@angular/forms/src/directives/ng_model';
import * as import5 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/di/injector';
import * as import7 from '@angular/core/src/linker/view_type';
import * as import8 from '@angular/core/src/change_detection/change_detection';
import * as import9 from '@angular/forms/src/directives/control_container';
import * as import10 from '@angular/core/src/linker/element_ref';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from '@angular/core/src/linker/component_factory';
import * as import13 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import14 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import15 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import16 from '@angular/common/src/directives/ng_class';
export class Wrapper_PagerComponent {
  context:import0.PagerComponent;
  changed:boolean;
  constructor(p0:any,p1:any,p2:any) {
    this.changed = false;
    this.context = new import0.PagerComponent(p0,p1,p2);
  }
  detectChangesInInputProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    return changed;
  }
  detectChangesInHostProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
}
var renderType_PagerComponent_Host:import2.RenderComponentType = (null as any);
class _View_PagerComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import3.AppElement;
  _NgModel_0_4:import4.NgModel;
  _PagerComponent_0_5:Wrapper_PagerComponent;
  constructor(viewUtils:import5.ViewUtils,parentInjector:import6.Injector,declarationEl:import3.AppElement) {
    super(_View_PagerComponent_Host0,renderType_PagerComponent_Host,import7.ViewType.HOST,viewUtils,parentInjector,declarationEl,import8.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import5.selectOrCreateRenderHostElement(this.renderer,'pager',new import5.InlineArray2(2,'ngModel',''),rootSelector,(null as any));
    this._appEl_0 = new import3.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_PagerComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._NgModel_0_4 = new import4.NgModel(this.parentInjector.get(import9.ControlContainer,(null as any)),(null as any),(null as any),(null as any));
    this._PagerComponent_0_5 = new Wrapper_PagerComponent(this._NgModel_0_4,this.renderer,new import10.ElementRef(this._el_0));
    this._appEl_0.initComponent(this._PagerComponent_0_5.context,([] as any[]),compView_0);
    compView_0.create(this._PagerComponent_0_5.context,this.projectableNodes,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import4.NgModel) && (0 === requestNodeIndex))) { return this._NgModel_0_4; }
    if (((token === import0.PagerComponent) && (0 === requestNodeIndex))) { return this._PagerComponent_0_5.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._PagerComponent_0_5.detectChangesInInputProps(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this._PagerComponent_0_5.detectChangesInHostProps(this,this._el_0,throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._NgModel_0_4.ngOnDestroy();
  }
}
function viewFactory_PagerComponent_Host0(viewUtils:import5.ViewUtils,parentInjector:import6.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  if ((renderType_PagerComponent_Host === (null as any))) { (renderType_PagerComponent_Host = viewUtils.createRenderComponentType('',0,import11.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_PagerComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const PagerComponentNgFactory:import12.ComponentFactory<import0.PagerComponent> = new import12.ComponentFactory<import0.PagerComponent>('pager[ngModel]',viewFactory_PagerComponent_Host0,import0.PagerComponent);
const styles_PagerComponent:any[] = ([] as any[]);
var renderType_PagerComponent:import2.RenderComponentType = (null as any);
class _View_PagerComponent0 extends import1.AppView<import0.PagerComponent> {
  _text_0:any;
  _el_1:any;
  _text_2:any;
  _el_3:any;
  _NgClass_3_3:import13.Wrapper_NgClass;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  _text_8:any;
  _el_9:any;
  _NgClass_9_3:import13.Wrapper_NgClass;
  _text_10:any;
  _el_11:any;
  _text_12:any;
  _text_13:any;
  _text_14:any;
  _text_15:any;
  /*private*/ _expr_18:any;
  /*private*/ _expr_19:any;
  _map_20:any;
  /*private*/ _expr_21:any;
  /*private*/ _expr_22:any;
  /*private*/ _expr_23:any;
  _map_24:any;
  /*private*/ _expr_25:any;
  constructor(viewUtils:import5.ViewUtils,parentInjector:import6.Injector,declarationEl:import3.AppElement) {
    super(_View_PagerComponent0,renderType_PagerComponent,import7.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import8.ChangeDetectorStatus.CheckAlways);
    this._expr_18 = import8.UNINITIALIZED;
    this._expr_19 = import8.UNINITIALIZED;
    this._map_20 = import5.pureProxy1((p0:any):{[key: string]:any} => {
      return {'pull-right': p0};
    });
    this._expr_21 = import8.UNINITIALIZED;
    this._expr_22 = import8.UNINITIALIZED;
    this._expr_23 = import8.UNINITIALIZED;
    this._map_24 = import5.pureProxy1((p0:any):{[key: string]:any} => {
      return {'pull-right': p0};
    });
    this._expr_25 = import8.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    this._el_1 = import5.createRenderElement(this.renderer,parentRenderNode,'ul',new import5.InlineArray2(2,'class','pager'),(null as any));
    this._text_2 = this.renderer.createText(this._el_1,'\n      ',(null as any));
    this._el_3 = import5.createRenderElement(this.renderer,this._el_1,'li',import5.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_3_3 = new import13.Wrapper_NgClass(this.parentInjector.get(import14.IterableDiffers),this.parentInjector.get(import15.KeyValueDiffers),new import10.ElementRef(this._el_3),this.renderer);
    this._text_4 = this.renderer.createText(this._el_3,'\n        ',(null as any));
    this._el_5 = import5.createRenderElement(this.renderer,this._el_3,'a',new import5.InlineArray2(2,'href',''),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'',(null as any));
    this._text_7 = this.renderer.createText(this._el_3,'\n      ',(null as any));
    this._text_8 = this.renderer.createText(this._el_1,'\n      ',(null as any));
    this._el_9 = import5.createRenderElement(this.renderer,this._el_1,'li',import5.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_9_3 = new import13.Wrapper_NgClass(this.parentInjector.get(import14.IterableDiffers),this.parentInjector.get(import15.KeyValueDiffers),new import10.ElementRef(this._el_9),this.renderer);
    this._text_10 = this.renderer.createText(this._el_9,'\n        ',(null as any));
    this._el_11 = import5.createRenderElement(this.renderer,this._el_9,'a',new import5.InlineArray2(2,'href',''),(null as any));
    this._text_12 = this.renderer.createText(this._el_11,'',(null as any));
    this._text_13 = this.renderer.createText(this._el_9,'\n      ',(null as any));
    this._text_14 = this.renderer.createText(this._el_1,'\n  ',(null as any));
    this._text_15 = this.renderer.createText(parentRenderNode,'\n',(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_5,'click',this.eventHandler(this._handle_click_5_0.bind(this)));
    var disposable_1:Function = this.renderer.listen(this._el_11,'click',this.eventHandler(this._handle_click_11_0.bind(this)));
    this.init(([] as any[]),[
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7,
      this._text_8,
      this._el_9,
      this._text_10,
      this._el_11,
      this._text_12,
      this._text_13,
      this._text_14,
      this._text_15
    ]
    ,[
      disposable_0,
      disposable_1
    ]
    ,([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.NgClass) && ((3 <= requestNodeIndex) && (requestNodeIndex <= 7)))) { return this._NgClass_3_3.context; }
    if (((token === import16.NgClass) && ((9 <= requestNodeIndex) && (requestNodeIndex <= 13)))) { return this._NgClass_9_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_3_0_0:any = import5.interpolate(1,'',this.context.pageBtnClass,'');
    this._NgClass_3_3.check_klass(currVal_3_0_0,throwOnChange,false);
    const currVal_3_0_1:any = this._map_20(this.context.align);
    this._NgClass_3_3.check_ngClass(currVal_3_0_1,throwOnChange,false);
    this._NgClass_3_3.detectChangesInInputProps(this,this._el_3,throwOnChange);
    const currVal_9_0_0:any = import5.interpolate(1,'',this.context.pageBtnClass,'');
    this._NgClass_9_3.check_klass(currVal_9_0_0,throwOnChange,false);
    const currVal_9_0_1:any = this._map_24(this.context.align);
    this._NgClass_9_3.check_ngClass(currVal_9_0_1,throwOnChange,false);
    this._NgClass_9_3.detectChangesInInputProps(this,this._el_9,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_18:any = this.context.noPrevious();
    if (import5.checkBinding(throwOnChange,this._expr_18,currVal_18)) {
      this.renderer.setElementClass(this._el_3,'disabled',currVal_18);
      this._expr_18 = currVal_18;
    }
    const currVal_19:any = this.context.align;
    if (import5.checkBinding(throwOnChange,this._expr_19,currVal_19)) {
      this.renderer.setElementClass(this._el_3,'previous',currVal_19);
      this._expr_19 = currVal_19;
    }
    this._NgClass_3_3.detectChangesInHostProps(this,this._el_3,throwOnChange);
    const currVal_21:any = import5.interpolate(1,'',this.context.getText('previous'),'');
    if (import5.checkBinding(throwOnChange,this._expr_21,currVal_21)) {
      this.renderer.setText(this._text_6,currVal_21);
      this._expr_21 = currVal_21;
    }
    const currVal_22:any = this.context.noNext();
    if (import5.checkBinding(throwOnChange,this._expr_22,currVal_22)) {
      this.renderer.setElementClass(this._el_9,'disabled',currVal_22);
      this._expr_22 = currVal_22;
    }
    const currVal_23:any = this.context.align;
    if (import5.checkBinding(throwOnChange,this._expr_23,currVal_23)) {
      this.renderer.setElementClass(this._el_9,'next',currVal_23);
      this._expr_23 = currVal_23;
    }
    this._NgClass_9_3.detectChangesInHostProps(this,this._el_9,throwOnChange);
    const currVal_25:any = import5.interpolate(1,'',this.context.getText('next'),'');
    if (import5.checkBinding(throwOnChange,this._expr_25,currVal_25)) {
      this.renderer.setText(this._text_12,currVal_25);
      this._expr_25 = currVal_25;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_5_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_5_0:any = ((<any>this.context.selectPage((this.context.page - 1),$event)) !== false);
    return (true && pd_5_0);
  }
  private _handle_click_11_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_11_0:any = ((<any>this.context.selectPage((this.context.page + 1),$event)) !== false);
    return (true && pd_11_0);
  }
}
export function viewFactory_PagerComponent0(viewUtils:import5.ViewUtils,parentInjector:import6.Injector,declarationEl:import3.AppElement):import1.AppView<import0.PagerComponent> {
  if ((renderType_PagerComponent === (null as any))) { (renderType_PagerComponent = viewUtils.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_PagerComponent,{})); }
  return new _View_PagerComponent0(viewUtils,parentInjector,declarationEl);
}
