// ng2-bootstrap/components/typeahead/typeahead-container.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/typeahead/typeahead-container.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/element';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from 'ng2-bootstrap/components/typeahead/typeahead-options.class';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
import * as import12 from '../../../@angular/common/src/directives/ng_style.ngfactory';
import * as import13 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import14 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import15 from '@angular/core/src/linker/template_ref';
import * as import16 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import17 from '@angular/common/src/directives/ng_for';
import * as import18 from '@angular/common/src/directives/ng_style';
import * as import19 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import20 from '@angular/common/src/directives/ng_if';
import * as import21 from '@angular/core/src/security';
import * as import22 from '../../../@angular/common/src/directives/ng_template_outlet.ngfactory';
import * as import23 from '@angular/common/src/directives/ng_template_outlet';
export class Wrapper_TypeaheadContainerComponent {
  context:import0.TypeaheadContainerComponent;
  changed:boolean;
  constructor(p0:any,p1:any) {
    this.changed = false;
    this.context = new import0.TypeaheadContainerComponent(p0,p1);
  }
  detectChangesInInputProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    return changed;
  }
  detectChangesInHostProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
}
var renderType_TypeaheadContainerComponent_Host:import2.RenderComponentType = (null as any);
class _View_TypeaheadContainerComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import3.AppElement;
  _TypeaheadContainerComponent_0_4:Wrapper_TypeaheadContainerComponent;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_TypeaheadContainerComponent_Host0,renderType_TypeaheadContainerComponent_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.selectOrCreateRenderHostElement(this.renderer,'typeahead-container',import4.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this._appEl_0 = new import3.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_TypeaheadContainerComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._TypeaheadContainerComponent_0_4 = new Wrapper_TypeaheadContainerComponent(new import8.ElementRef(this._el_0),this.parentInjector.get(import9.TypeaheadOptions));
    this._appEl_0.initComponent(this._TypeaheadContainerComponent_0_4.context,([] as any[]),compView_0);
    compView_0.create(this._TypeaheadContainerComponent_0_4.context,this.projectableNodes,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.TypeaheadContainerComponent) && (0 === requestNodeIndex))) { return this._TypeaheadContainerComponent_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._TypeaheadContainerComponent_0_4.detectChangesInInputProps(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this._TypeaheadContainerComponent_0_4.detectChangesInHostProps(this,this._el_0,throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_TypeaheadContainerComponent_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  if ((renderType_TypeaheadContainerComponent_Host === (null as any))) { (renderType_TypeaheadContainerComponent_Host = viewUtils.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_TypeaheadContainerComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const TypeaheadContainerComponentNgFactory:import11.ComponentFactory<import0.TypeaheadContainerComponent> = new import11.ComponentFactory<import0.TypeaheadContainerComponent>('typeahead-container',viewFactory_TypeaheadContainerComponent_Host0,import0.TypeaheadContainerComponent);
const styles_TypeaheadContainerComponent:any[] = ([] as any[]);
var renderType_TypeaheadContainerComponent:import2.RenderComponentType = (null as any);
class _View_TypeaheadContainerComponent0 extends import1.AppView<import0.TypeaheadContainerComponent> {
  _text_0:any;
  _el_1:any;
  _NgStyle_1_3:import12.Wrapper_NgStyle;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _appEl_3:import3.AppElement;
  _TemplateRef_3_5:any;
  _NgFor_3_6:import13.Wrapper_NgFor;
  _text_4:any;
  _text_5:any;
  _map_10:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_TypeaheadContainerComponent0,renderType_TypeaheadContainerComponent,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._map_10 = import4.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        top: p0,
        left: p1,
        display: p2
      }
      ;
    });
  }
  createInternal(rootSelector:string):import3.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this._el_1 = import4.createRenderElement(this.renderer,parentRenderNode,'ul',new import4.InlineArray2(2,'class','dropdown-menu'),(null as any));
    this._NgStyle_1_3 = new import12.Wrapper_NgStyle(this.parentInjector.get(import14.KeyValueDiffers),new import8.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText(this._el_1,'\n    ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor(this._el_1,(null as any));
    this._appEl_3 = new import3.AppElement(3,1,this,this._anchor_3);
    this._TemplateRef_3_5 = new import15.TemplateRef_(this._appEl_3,viewFactory_TypeaheadContainerComponent1);
    this._NgFor_3_6 = new import13.Wrapper_NgFor(this._appEl_3.vcRef,this._TemplateRef_3_5,this.parentInjector.get(import16.IterableDiffers),this.ref);
    this._text_4 = this.renderer.createText(this._el_1,'\n  ',(null as any));
    this._text_5 = this.renderer.createText(parentRenderNode,'\n',(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_1,'mouseleave',this.eventHandler(this._handle_mouseleave_1_0.bind(this)));
    this.init(([] as any[]),[
      this._text_0,
      this._el_1,
      this._text_2,
      this._anchor_3,
      this._text_4,
      this._text_5
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import15.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import17.NgFor) && (3 === requestNodeIndex))) { return this._NgFor_3_6.context; }
    if (((token === import18.NgStyle) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgStyle_1_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this._map_10(this.context.top,this.context.left,'block');
    this._NgStyle_1_3.check_ngStyle(currVal_1_0_0,throwOnChange,false);
    this._NgStyle_1_3.detectChangesInInputProps(this,this._el_1,throwOnChange);
    const currVal_3_0_0:any = this.context.matches;
    this._NgFor_3_6.check_ngForOf(currVal_3_0_0,throwOnChange,false);
    this._NgFor_3_6.detectChangesInInputProps(this,this._anchor_3,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this._NgStyle_1_3.detectChangesInHostProps(this,this._el_1,throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_mouseleave_1_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_1_0:any = ((<any>this.context.focusLost()) !== false);
    return (true && pd_1_0);
  }
}
export function viewFactory_TypeaheadContainerComponent0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<import0.TypeaheadContainerComponent> {
  if ((renderType_TypeaheadContainerComponent === (null as any))) { (renderType_TypeaheadContainerComponent = viewUtils.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_TypeaheadContainerComponent,{})); }
  return new _View_TypeaheadContainerComponent0(viewUtils,parentInjector,declarationEl);
}
class _View_TypeaheadContainerComponent1 extends import1.AppView<any> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _appEl_1:import3.AppElement;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import19.Wrapper_NgIf;
  _text_2:any;
  _anchor_3:any;
  /*private*/ _appEl_3:import3.AppElement;
  _TemplateRef_3_5:any;
  _NgIf_3_6:import19.Wrapper_NgIf;
  _text_4:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_TypeaheadContainerComponent1,renderType_TypeaheadContainerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._text_0 = this.renderer.createText((null as any),'\n      ',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._appEl_1 = new import3.AppElement(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import15.TemplateRef_(this._appEl_1,viewFactory_TypeaheadContainerComponent2);
    this._NgIf_1_6 = new import19.Wrapper_NgIf(this._appEl_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText((null as any),'\n      ',(null as any));
    this._anchor_3 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._appEl_3 = new import3.AppElement(3,(null as any),this,this._anchor_3);
    this._TemplateRef_3_5 = new import15.TemplateRef_(this._appEl_3,viewFactory_TypeaheadContainerComponent3);
    this._NgIf_3_6 = new import19.Wrapper_NgIf(this._appEl_3.vcRef,this._TemplateRef_3_5);
    this._text_4 = this.renderer.createText((null as any),'\n    ',(null as any));
    this.init(([] as any[]).concat([
      this._text_0,
      this._appEl_1,
      this._text_2,
      this._appEl_3,
      this._text_4
    ]
    ),[
      this._text_0,
      this._anchor_1,
      this._text_2,
      this._anchor_3,
      this._text_4
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import15.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import20.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    if (((token === import15.TemplateRef) && (3 === requestNodeIndex))) { return this._TemplateRef_3_5; }
    if (((token === import20.NgIf) && (3 === requestNodeIndex))) { return this._NgIf_3_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = this.context.$implicit.isHeader();
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.detectChangesInInputProps(this,this._anchor_1,throwOnChange);
    const currVal_3_0_0:boolean = !this.context.$implicit.isHeader();
    this._NgIf_3_6.check_ngIf(currVal_3_0_0,throwOnChange,false);
    this._NgIf_3_6.detectChangesInInputProps(this,this._anchor_3,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_TypeaheadContainerComponent1(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_TypeaheadContainerComponent1(viewUtils,parentInjector,declarationEl);
}
class _View_TypeaheadContainerComponent2 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  /*private*/ _expr_2:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_TypeaheadContainerComponent2,renderType_TypeaheadContainerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_2 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'li',new import4.InlineArray2(2,'class','dropdown-header'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'',(null as any));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_2:any = import4.interpolate(1,'',this.parent.context.$implicit,'');
    if (import4.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setText(this._text_1,currVal_2);
      this._expr_2 = currVal_2;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_TypeaheadContainerComponent2(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_TypeaheadContainerComponent2(viewUtils,parentInjector,declarationEl);
}
class _View_TypeaheadContainerComponent3 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _appEl_2:import3.AppElement;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import19.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _appEl_4:import3.AppElement;
  _TemplateRef_4_5:any;
  _NgIf_4_6:import19.Wrapper_NgIf;
  _text_5:any;
  /*private*/ _expr_12:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_TypeaheadContainerComponent3,renderType_TypeaheadContainerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_12 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'li',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._appEl_2 = new import3.AppElement(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import15.TemplateRef_(this._appEl_2,viewFactory_TypeaheadContainerComponent4);
    this._NgIf_2_6 = new import19.Wrapper_NgIf(this._appEl_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._appEl_4 = new import3.AppElement(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import15.TemplateRef_(this._appEl_4,viewFactory_TypeaheadContainerComponent5);
    this._NgIf_4_6 = new import19.Wrapper_NgIf(this._appEl_4.vcRef,this._TemplateRef_4_5);
    this._text_5 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_0,'mouseenter',this.eventHandler(this._handle_mouseenter_0_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import15.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import20.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import15.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import20.NgIf) && (4 === requestNodeIndex))) { return this._NgIf_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:boolean = !this.parent.parent.context.itemTemplate;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.detectChangesInInputProps(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:any = this.parent.parent.context.itemTemplate;
    this._NgIf_4_6.check_ngIf(currVal_4_0_0,throwOnChange,false);
    this._NgIf_4_6.detectChangesInInputProps(this,this._anchor_4,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_12:any = this.parent.parent.context.isActive(this.parent.context.$implicit);
    if (import4.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this.renderer.setElementClass(this._el_0,'active',currVal_12);
      this._expr_12 = currVal_12;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_mouseenter_0_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0_0:any = ((<any>this.parent.parent.context.selectActive(this.parent.context.$implicit)) !== false);
    return (true && pd_0_0);
  }
}
function viewFactory_TypeaheadContainerComponent3(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_TypeaheadContainerComponent3(viewUtils,parentInjector,declarationEl);
}
class _View_TypeaheadContainerComponent4 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _expr_1:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_TypeaheadContainerComponent4,renderType_TypeaheadContainerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_1 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'a',new import4.InlineArray4(4,'href','#','tabindex','-1'),(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_0,'click',this.eventHandler(this._handle_click_0_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],[disposable_0],([] as any[]));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_1:any = this.parent.parent.parent.context.hightlight(this.parent.parent.context.$implicit,this.parent.parent.parent.context.query);
    if (import4.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementProperty(this._el_0,'innerHTML',this.viewUtils.sanitizer.sanitize(import21.SecurityContext.HTML,currVal_1));
      this._expr_1 = currVal_1;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_0_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0_0:any = ((<any>this.parent.parent.parent.context.selectMatch(this.parent.parent.context.$implicit,$event)) !== false);
    return (true && pd_0_0);
  }
}
function viewFactory_TypeaheadContainerComponent4(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_TypeaheadContainerComponent4(viewUtils,parentInjector,declarationEl);
}
class _View_TypeaheadContainerComponent5 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _appEl_2:import3.AppElement;
  _TemplateRef_2_5:any;
  _NgTemplateOutlet_2_6:import22.Wrapper_NgTemplateOutlet;
  _text_3:any;
  _map_7:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_TypeaheadContainerComponent5,renderType_TypeaheadContainerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._map_7 = import4.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        item: p0,
        index: p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'a',new import4.InlineArray4(4,'href','#','tabindex','-1'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._appEl_2 = new import3.AppElement(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import15.TemplateRef_(this._appEl_2,viewFactory_TypeaheadContainerComponent6);
    this._NgTemplateOutlet_2_6 = new import22.Wrapper_NgTemplateOutlet(this._appEl_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_0,'click',this.eventHandler(this._handle_click_0_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import15.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import23.NgTemplateOutlet) && (2 === requestNodeIndex))) { return this._NgTemplateOutlet_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this._map_7(this.parent.parent.context.$implicit.item,this.parent.parent.context.index);
    this._NgTemplateOutlet_2_6.check_ngOutletContext(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parent.parent.parent.context.itemTemplate;
    this._NgTemplateOutlet_2_6.check_ngTemplateOutlet(currVal_2_0_1,throwOnChange,false);
    this._NgTemplateOutlet_2_6.detectChangesInInputProps(this,this._anchor_2,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_0_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0_0:any = ((<any>this.parent.parent.parent.context.selectMatch(this.parent.parent.context.$implicit,$event)) !== false);
    return (true && pd_0_0);
  }
}
function viewFactory_TypeaheadContainerComponent5(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_TypeaheadContainerComponent5(viewUtils,parentInjector,declarationEl);
}
class _View_TypeaheadContainerComponent6 extends import1.AppView<any> {
  _text_0:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_TypeaheadContainerComponent6,renderType_TypeaheadContainerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._text_0 = this.renderer.createText((null as any),'\n            ',(null as any));
    this.init(([] as any[]).concat([this._text_0]),[this._text_0],([] as any[]),([] as any[]));
    return (null as any);
  }
}
function viewFactory_TypeaheadContainerComponent6(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_TypeaheadContainerComponent6(viewUtils,parentInjector,declarationEl);
}
