// ng2-bootstrap/components/datepicker/daypicker.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/datepicker/daypicker.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/element';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ng2-bootstrap/components/datepicker/datepicker-inner.component';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
import * as import11 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import12 from '@angular/core/src/linker/template_ref';
import * as import13 from '@angular/common/src/directives/ng_if';
import * as import14 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import15 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import16 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import17 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import18 from '@angular/core/src/linker/element_ref';
import * as import19 from '@angular/common/src/directives/ng_class';
import * as import20 from '@angular/common/src/directives/ng_for';
import * as import21 from '@angular/core/src/security';
export class Wrapper_DayPickerComponent {
  context:import0.DayPickerComponent;
  changed:boolean;
  constructor(p0:any) {
    this.changed = false;
    this.context = new import0.DayPickerComponent(p0);
  }
  detectChangesInInputProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  detectChangesInHostProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
}
var renderType_DayPickerComponent_Host:import2.RenderComponentType = (null as any);
class _View_DayPickerComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import3.AppElement;
  _DayPickerComponent_0_4:Wrapper_DayPickerComponent;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent_Host0,renderType_DayPickerComponent_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.selectOrCreateRenderHostElement(this.renderer,'daypicker',import4.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this._appEl_0 = new import3.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_DayPickerComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._DayPickerComponent_0_4 = new Wrapper_DayPickerComponent(this.parentInjector.get(import8.DatePickerInnerComponent));
    this._appEl_0.initComponent(this._DayPickerComponent_0_4.context,([] as any[]),compView_0);
    compView_0.create(this._DayPickerComponent_0_4.context,this.projectableNodes,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.DayPickerComponent) && (0 === requestNodeIndex))) { return this._DayPickerComponent_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._DayPickerComponent_0_4.detectChangesInInputProps(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this._DayPickerComponent_0_4.detectChangesInHostProps(this,this._el_0,throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_DayPickerComponent_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  if ((renderType_DayPickerComponent_Host === (null as any))) { (renderType_DayPickerComponent_Host = viewUtils.createRenderComponentType('',0,import9.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_DayPickerComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const DayPickerComponentNgFactory:import10.ComponentFactory<import0.DayPickerComponent> = new import10.ComponentFactory<import0.DayPickerComponent>('daypicker',viewFactory_DayPickerComponent_Host0,import0.DayPickerComponent);
const styles_DayPickerComponent:any[] = ([] as any[]);
var renderType_DayPickerComponent:import2.RenderComponentType = (null as any);
class _View_DayPickerComponent0 extends import1.AppView<import0.DayPickerComponent> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _appEl_1:import3.AppElement;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import11.Wrapper_NgIf;
  _text_2:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent0,renderType_DayPickerComponent,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._appEl_1 = new import3.AppElement(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import12.TemplateRef_(this._appEl_1,viewFactory_DayPickerComponent1);
    this._NgIf_1_6 = new import11.Wrapper_NgIf(this._appEl_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this.init(([] as any[]),[
      this._text_0,
      this._anchor_1,
      this._text_2
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import13.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = (this.context.datePicker.datepickerMode === 'day');
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.detectChangesInInputProps(this,this._anchor_1,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
export function viewFactory_DayPickerComponent0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<import0.DayPickerComponent> {
  if ((renderType_DayPickerComponent === (null as any))) { (renderType_DayPickerComponent = viewUtils.createRenderComponentType('',0,import9.ViewEncapsulation.None,styles_DayPickerComponent,{})); }
  return new _View_DayPickerComponent0(viewUtils,parentInjector,declarationEl);
}
class _View_DayPickerComponent1 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _el_8:any;
  _text_9:any;
  _text_10:any;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _el_14:any;
  _NgClass_14_3:import14.Wrapper_NgClass;
  _text_15:any;
  _el_16:any;
  _text_17:any;
  _text_18:any;
  _text_19:any;
  _text_20:any;
  _el_21:any;
  _text_22:any;
  _el_23:any;
  _text_24:any;
  _text_25:any;
  _text_26:any;
  _text_27:any;
  _el_28:any;
  _text_29:any;
  _anchor_30:any;
  /*private*/ _appEl_30:import3.AppElement;
  _TemplateRef_30_5:any;
  _NgIf_30_6:import11.Wrapper_NgIf;
  _text_31:any;
  _anchor_32:any;
  /*private*/ _appEl_32:import3.AppElement;
  _TemplateRef_32_5:any;
  _NgFor_32_6:import15.Wrapper_NgFor;
  _text_33:any;
  _text_34:any;
  _text_35:any;
  _el_36:any;
  _text_37:any;
  _anchor_38:any;
  /*private*/ _appEl_38:import3.AppElement;
  _TemplateRef_38_5:any;
  _NgFor_38_6:import15.Wrapper_NgFor;
  _text_39:any;
  _text_40:any;
  /*private*/ _expr_51:any;
  /*private*/ _expr_52:any;
  /*private*/ _expr_53:any;
  /*private*/ _expr_54:any;
  /*private*/ _expr_55:any;
  _map_56:any;
  /*private*/ _expr_57:any;
  /*private*/ _expr_58:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent1,renderType_DayPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_51 = import7.UNINITIALIZED;
    this._expr_52 = import7.UNINITIALIZED;
    this._expr_53 = import7.UNINITIALIZED;
    this._expr_54 = import7.UNINITIALIZED;
    this._expr_55 = import7.UNINITIALIZED;
    this._map_56 = import4.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._expr_57 = import7.UNINITIALIZED;
    this._expr_58 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'table',new import4.InlineArray4(4,'aria-activedescendant','activeDateId','role','grid'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n  ',(null as any));
    this._el_2 = import4.createRenderElement(this.renderer,this._el_0,'thead',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n    ',(null as any));
    this._el_4 = import4.createRenderElement(this.renderer,this._el_2,'tr',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_6 = import4.createRenderElement(this.renderer,this._el_4,'th',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_7 = this.renderer.createText(this._el_6,'\n        ',(null as any));
    this._el_8 = import4.createRenderElement(this.renderer,this._el_6,'button',new import4.InlineArray8(6,'class','btn btn-default btn-secondary btn-sm pull-left','tabindex','-1','type','button'),(null as any));
    this._text_9 = this.renderer.createText(this._el_8,'\n        ',(null as any));
    this._text_10 = this.renderer.createText(this._el_6,'\n      ',(null as any));
    this._text_11 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_12 = import4.createRenderElement(this.renderer,this._el_4,'th',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_13 = this.renderer.createText(this._el_12,'\n        ',(null as any));
    this._el_14 = import4.createRenderElement(this.renderer,this._el_12,'button',new import4.InlineArray8(8,'class','btn btn-default btn-secondary btn-sm','style','width:100%;','tabindex','-1','type','button'),(null as any));
    this._NgClass_14_3 = new import14.Wrapper_NgClass(this.parentInjector.get(import16.IterableDiffers),this.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_14),this.renderer);
    this._text_15 = this.renderer.createText(this._el_14,'\n          ',(null as any));
    this._el_16 = import4.createRenderElement(this.renderer,this._el_14,'strong',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_17 = this.renderer.createText(this._el_16,'',(null as any));
    this._text_18 = this.renderer.createText(this._el_14,'\n        ',(null as any));
    this._text_19 = this.renderer.createText(this._el_12,'\n      ',(null as any));
    this._text_20 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_21 = import4.createRenderElement(this.renderer,this._el_4,'th',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_22 = this.renderer.createText(this._el_21,'\n        ',(null as any));
    this._el_23 = import4.createRenderElement(this.renderer,this._el_21,'button',new import4.InlineArray8(6,'class','btn btn-default btn-secondary btn-sm pull-right','tabindex','-1','type','button'),(null as any));
    this._text_24 = this.renderer.createText(this._el_23,'\n        ',(null as any));
    this._text_25 = this.renderer.createText(this._el_21,'\n      ',(null as any));
    this._text_26 = this.renderer.createText(this._el_4,'\n    ',(null as any));
    this._text_27 = this.renderer.createText(this._el_2,'\n    ',(null as any));
    this._el_28 = import4.createRenderElement(this.renderer,this._el_2,'tr',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_29 = this.renderer.createText(this._el_28,'\n      ',(null as any));
    this._anchor_30 = this.renderer.createTemplateAnchor(this._el_28,(null as any));
    this._appEl_30 = new import3.AppElement(30,28,this,this._anchor_30);
    this._TemplateRef_30_5 = new import12.TemplateRef_(this._appEl_30,viewFactory_DayPickerComponent2);
    this._NgIf_30_6 = new import11.Wrapper_NgIf(this._appEl_30.vcRef,this._TemplateRef_30_5);
    this._text_31 = this.renderer.createText(this._el_28,'\n      ',(null as any));
    this._anchor_32 = this.renderer.createTemplateAnchor(this._el_28,(null as any));
    this._appEl_32 = new import3.AppElement(32,28,this,this._anchor_32);
    this._TemplateRef_32_5 = new import12.TemplateRef_(this._appEl_32,viewFactory_DayPickerComponent3);
    this._NgFor_32_6 = new import15.Wrapper_NgFor(this._appEl_32.vcRef,this._TemplateRef_32_5,this.parentInjector.get(import16.IterableDiffers),this.parent.ref);
    this._text_33 = this.renderer.createText(this._el_28,'\n    ',(null as any));
    this._text_34 = this.renderer.createText(this._el_2,'\n  ',(null as any));
    this._text_35 = this.renderer.createText(this._el_0,'\n  ',(null as any));
    this._el_36 = import4.createRenderElement(this.renderer,this._el_0,'tbody',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_37 = this.renderer.createText(this._el_36,'\n    ',(null as any));
    this._anchor_38 = this.renderer.createTemplateAnchor(this._el_36,(null as any));
    this._appEl_38 = new import3.AppElement(38,36,this,this._anchor_38);
    this._TemplateRef_38_5 = new import12.TemplateRef_(this._appEl_38,viewFactory_DayPickerComponent4);
    this._NgFor_38_6 = new import15.Wrapper_NgFor(this._appEl_38.vcRef,this._TemplateRef_38_5,this.parentInjector.get(import16.IterableDiffers),this.parent.ref);
    this._text_39 = this.renderer.createText(this._el_36,'\n  ',(null as any));
    this._text_40 = this.renderer.createText(this._el_0,'\n',(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_8,'click',this.eventHandler(this._handle_click_8_0.bind(this)));
    var disposable_1:Function = this.renderer.listen(this._el_14,'click',this.eventHandler(this._handle_click_14_0.bind(this)));
    var disposable_2:Function = this.renderer.listen(this._el_23,'click',this.eventHandler(this._handle_click_23_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._text_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._el_14,
      this._text_15,
      this._el_16,
      this._text_17,
      this._text_18,
      this._text_19,
      this._text_20,
      this._el_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._text_25,
      this._text_26,
      this._text_27,
      this._el_28,
      this._text_29,
      this._anchor_30,
      this._text_31,
      this._anchor_32,
      this._text_33,
      this._text_34,
      this._text_35,
      this._el_36,
      this._text_37,
      this._anchor_38,
      this._text_39,
      this._text_40
    ]
    ,[
      disposable_0,
      disposable_1,
      disposable_2
    ]
    ,([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.NgClass) && ((14 <= requestNodeIndex) && (requestNodeIndex <= 18)))) { return this._NgClass_14_3.context; }
    if (((token === import12.TemplateRef) && (30 === requestNodeIndex))) { return this._TemplateRef_30_5; }
    if (((token === import13.NgIf) && (30 === requestNodeIndex))) { return this._NgIf_30_6.context; }
    if (((token === import12.TemplateRef) && (32 === requestNodeIndex))) { return this._TemplateRef_32_5; }
    if (((token === import20.NgFor) && (32 === requestNodeIndex))) { return this._NgFor_32_6.context; }
    if (((token === import12.TemplateRef) && (38 === requestNodeIndex))) { return this._TemplateRef_38_5; }
    if (((token === import20.NgFor) && (38 === requestNodeIndex))) { return this._NgFor_38_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_14_0_0:any = 'btn btn-default btn-secondary btn-sm';
    this._NgClass_14_3.check_klass(currVal_14_0_0,throwOnChange,false);
    const currVal_14_0_1:any = this._map_56((this.parent.context.datePicker.datepickerMode === this.parent.context.datePicker.maxMode));
    this._NgClass_14_3.check_ngClass(currVal_14_0_1,throwOnChange,false);
    this._NgClass_14_3.detectChangesInInputProps(this,this._el_14,throwOnChange);
    const currVal_30_0_0:any = this.parent.context.datePicker.showWeeks;
    this._NgIf_30_6.check_ngIf(currVal_30_0_0,throwOnChange,false);
    this._NgIf_30_6.detectChangesInInputProps(this,this._anchor_30,throwOnChange);
    const currVal_32_0_0:any = this.parent.context.labels;
    this._NgFor_32_6.check_ngForOf(currVal_32_0_0,throwOnChange,false);
    this._NgFor_32_6.detectChangesInInputProps(this,this._anchor_32,throwOnChange);
    const currVal_38_0_0:any = this.parent.context.rows;
    this._NgFor_38_6.check_ngForOf(currVal_38_0_0,throwOnChange,false);
    this._NgFor_38_6.detectChangesInInputProps(this,this._anchor_38,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_51:any = (this.parent.context.datePicker.uniqueId + '-title');
    if (import4.checkBinding(throwOnChange,this._expr_51,currVal_51)) {
      this.renderer.setElementAttribute(this._el_0,'aria-labelledby',((currVal_51 == (null as any))? (null as any): currVal_51.toString()));
      this._expr_51 = currVal_51;
    }
    const currVal_52:any = this.parent.context.CURRENT_THEME_TEMPLATE.ARROW_LEFT;
    if (import4.checkBinding(throwOnChange,this._expr_52,currVal_52)) {
      this.renderer.setElementProperty(this._el_8,'innerHTML',this.viewUtils.sanitizer.sanitize(import21.SecurityContext.HTML,currVal_52));
      this._expr_52 = currVal_52;
    }
    const currVal_53:any = (5 + (this.parent.context.datePicker.showWeeks? 1: 0));
    if (import4.checkBinding(throwOnChange,this._expr_53,currVal_53)) {
      this.renderer.setElementAttribute(this._el_12,'colspan',((currVal_53 == (null as any))? (null as any): currVal_53.toString()));
      this._expr_53 = currVal_53;
    }
    const currVal_54:any = (this.parent.context.datePicker.uniqueId + '-title');
    if (import4.checkBinding(throwOnChange,this._expr_54,currVal_54)) {
      this.renderer.setElementProperty(this._el_14,'id',currVal_54);
      this._expr_54 = currVal_54;
    }
    const currVal_55:any = (this.parent.context.datePicker.datepickerMode === this.parent.context.datePicker.maxMode);
    if (import4.checkBinding(throwOnChange,this._expr_55,currVal_55)) {
      this.renderer.setElementProperty(this._el_14,'disabled',currVal_55);
      this._expr_55 = currVal_55;
    }
    this._NgClass_14_3.detectChangesInHostProps(this,this._el_14,throwOnChange);
    const currVal_57:any = import4.interpolate(1,'',this.parent.context.title,'');
    if (import4.checkBinding(throwOnChange,this._expr_57,currVal_57)) {
      this.renderer.setText(this._text_17,currVal_57);
      this._expr_57 = currVal_57;
    }
    const currVal_58:any = this.parent.context.CURRENT_THEME_TEMPLATE.ARROW_RIGHT;
    if (import4.checkBinding(throwOnChange,this._expr_58,currVal_58)) {
      this.renderer.setElementProperty(this._el_23,'innerHTML',this.viewUtils.sanitizer.sanitize(import21.SecurityContext.HTML,currVal_58));
      this._expr_58 = currVal_58;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_8_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_8_0:any = ((<any>this.parent.context.datePicker.move((0 - 1))) !== false);
    return (true && pd_8_0);
  }
  private _handle_click_14_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_14_0:any = ((<any>this.parent.context.datePicker.toggleMode()) !== false);
    return (true && pd_14_0);
  }
  private _handle_click_23_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_23_0:any = ((<any>this.parent.context.datePicker.move(1)) !== false);
    return (true && pd_23_0);
  }
}
function viewFactory_DayPickerComponent1(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_DayPickerComponent1(viewUtils,parentInjector,declarationEl);
}
class _View_DayPickerComponent2 extends import1.AppView<any> {
  _el_0:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent2,renderType_DayPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'th',import4.EMPTY_INLINE_ARRAY,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return (null as any);
  }
}
function viewFactory_DayPickerComponent2(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_DayPickerComponent2(viewUtils,parentInjector,declarationEl);
}
class _View_DayPickerComponent3 extends import1.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import14.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _el_3:any;
  _text_4:any;
  _text_5:any;
  _map_7:any;
  /*private*/ _expr_8:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent3,renderType_DayPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._map_7 = import4.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'text-xs-center': p0,
        'text-center': p1
      }
      ;
    });
    this._expr_8 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'th',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import14.Wrapper_NgClass(this.parent.parentInjector.get(import16.IterableDiffers),this.parent.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._el_2 = import4.createRenderElement(this.renderer,this._el_0,'small',new import4.InlineArray2(2,'aria-label','labelz.full'),(null as any));
    this._el_3 = import4.createRenderElement(this.renderer,this._el_2,'b',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_4 = this.renderer.createText(this._el_3,'',(null as any));
    this._text_5 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._el_3,
      this._text_4,
      this._text_5
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_7(this.parent.parent.context.isBS4,!this.parent.parent.context.isBS4);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.detectChangesInInputProps(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this._NgClass_0_3.detectChangesInHostProps(this,this._el_0,throwOnChange);
    const currVal_8:any = import4.interpolate(1,'',this.context.$implicit.abbr,'');
    if (import4.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setText(this._text_4,currVal_8);
      this._expr_8 = currVal_8;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_DayPickerComponent3(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_DayPickerComponent3(viewUtils,parentInjector,declarationEl);
}
class _View_DayPickerComponent4 extends import1.AppView<any> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _appEl_1:import3.AppElement;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import11.Wrapper_NgIf;
  _text_2:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent4,renderType_DayPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._text_0 = this.renderer.createText((null as any),'\n      ',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._appEl_1 = new import3.AppElement(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import12.TemplateRef_(this._appEl_1,viewFactory_DayPickerComponent5);
    this._NgIf_1_6 = new import11.Wrapper_NgIf(this._appEl_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText((null as any),'\n    ',(null as any));
    this.init(([] as any[]).concat([
      this._text_0,
      this._appEl_1,
      this._text_2
    ]
    ),[
      this._text_0,
      this._anchor_1,
      this._text_2
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import13.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:boolean = !((this.parent.parent.context.datePicker.onlyCurrentMonth && this.context.$implicit[0].secondary) && this.context.$implicit[6].secondary);
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.detectChangesInInputProps(this,this._anchor_1,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_DayPickerComponent4(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_DayPickerComponent4(viewUtils,parentInjector,declarationEl);
}
class _View_DayPickerComponent5 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _appEl_2:import3.AppElement;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import11.Wrapper_NgIf;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _appEl_4:import3.AppElement;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import15.Wrapper_NgFor;
  _text_5:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent5,renderType_DayPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'tr',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._appEl_2 = new import3.AppElement(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import12.TemplateRef_(this._appEl_2,viewFactory_DayPickerComponent6);
    this._NgIf_2_6 = new import11.Wrapper_NgIf(this._appEl_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._appEl_4 = new import3.AppElement(4,0,this,this._anchor_4);
    this._TemplateRef_4_5 = new import12.TemplateRef_(this._appEl_4,viewFactory_DayPickerComponent7);
    this._NgFor_4_6 = new import15.Wrapper_NgFor(this._appEl_4.vcRef,this._TemplateRef_4_5,this.parent.parent.parentInjector.get(import16.IterableDiffers),this.parent.parent.parent.ref);
    this._text_5 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3,
      this._anchor_4,
      this._text_5
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import13.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import12.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import20.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.parent.parent.parent.context.datePicker.showWeeks;
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.detectChangesInInputProps(this,this._anchor_2,throwOnChange);
    const currVal_4_0_0:any = this.parent.context.$implicit;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.detectChangesInInputProps(this,this._anchor_4,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_DayPickerComponent5(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_DayPickerComponent5(viewUtils,parentInjector,declarationEl);
}
class _View_DayPickerComponent6 extends import1.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import14.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _map_6:any;
  /*private*/ _expr_7:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent6,renderType_DayPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._map_6 = import4.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'text-xs-center': p0,
        'text-center': p1
      }
      ;
    });
    this._expr_7 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'td',new import4.InlineArray2(2,'class','h6'),(null as any));
    this._NgClass_0_3 = new import14.Wrapper_NgClass(this.parent.parent.parent.parentInjector.get(import16.IterableDiffers),this.parent.parent.parent.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n          ',(null as any));
    this._el_2 = import4.createRenderElement(this.renderer,this._el_0,'em',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'h6';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_6(this.parent.parent.parent.parent.context.isBS4,!this.parent.parent.parent.parent.context.isBS4);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.detectChangesInInputProps(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this._NgClass_0_3.detectChangesInHostProps(this,this._el_0,throwOnChange);
    const currVal_7:any = import4.interpolate(1,'',this.parent.parent.parent.parent.context.weekNumbers[this.parent.parent.context.index],'');
    if (import4.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setText(this._text_3,currVal_7);
      this._expr_7 = currVal_7;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_DayPickerComponent6(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_DayPickerComponent6(viewUtils,parentInjector,declarationEl);
}
class _View_DayPickerComponent7 extends import1.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import14.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _appEl_2:import3.AppElement;
  _TemplateRef_2_5:any;
  _NgIf_2_6:import11.Wrapper_NgIf;
  _text_3:any;
  /*private*/ _expr_8:any;
  _map_9:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent7,renderType_DayPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_8 = import7.UNINITIALIZED;
    this._map_9 = import4.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'text-xs-center': p0,
        'text-center': p1
      }
      ;
    });
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'td',new import4.InlineArray2(2,'role','gridcell'),(null as any));
    this._NgClass_0_3 = new import14.Wrapper_NgClass(this.parent.parent.parent.parentInjector.get(import16.IterableDiffers),this.parent.parent.parent.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n          ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._appEl_2 = new import3.AppElement(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import12.TemplateRef_(this._appEl_2,viewFactory_DayPickerComponent8);
    this._NgIf_2_6 = new import11.Wrapper_NgIf(this._appEl_2.vcRef,this._TemplateRef_2_5);
    this._text_3 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import13.NgIf) && (2 === requestNodeIndex))) { return this._NgIf_2_6.context; }
    if (((token === import19.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_9(this.parent.parent.parent.parent.context.isBS4,!this.parent.parent.parent.parent.context.isBS4);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.detectChangesInInputProps(this,this._el_0,throwOnChange);
    const currVal_2_0_0:boolean = !(this.parent.parent.parent.parent.context.datePicker.onlyCurrentMonth && this.context.$implicit.secondary);
    this._NgIf_2_6.check_ngIf(currVal_2_0_0,throwOnChange,false);
    this._NgIf_2_6.detectChangesInInputProps(this,this._anchor_2,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_8:any = this.context.$implicit.uid;
    if (import4.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this.renderer.setElementProperty(this._el_0,'id',currVal_8);
      this._expr_8 = currVal_8;
    }
    this._NgClass_0_3.detectChangesInHostProps(this,this._el_0,throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_DayPickerComponent7(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_DayPickerComponent7(viewUtils,parentInjector,declarationEl);
}
class _View_DayPickerComponent8 extends import1.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import14.Wrapper_NgClass;
  _text_1:any;
  _el_2:any;
  _NgClass_2_3:import14.Wrapper_NgClass;
  _text_3:any;
  _text_4:any;
  /*private*/ _expr_7:any;
  _map_8:any;
  _map_9:any;
  /*private*/ _expr_10:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_DayPickerComponent8,renderType_DayPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_7 = import7.UNINITIALIZED;
    this._map_8 = import4.pureProxy5((p0:any,p1:any,p2:any,p3:any,p4:any):{[key: string]:any} => {
      return {
        'btn-secondary': p0,
        'btn-info': p1,
        disabled: p2,
        active: p3,
        'btn-default': p4
      }
      ;
    });
    this._map_9 = import4.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'text-muted': p0,
        'text-info': p1
      }
      ;
    });
    this._expr_10 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'button',new import4.InlineArray8(6,'style','min-width:100%;','tabindex','-1','type','button'),(null as any));
    this._NgClass_0_3 = new import14.Wrapper_NgClass(this.parent.parent.parent.parent.parentInjector.get(import16.IterableDiffers),this.parent.parent.parent.parent.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n            ',(null as any));
    this._el_2 = import4.createRenderElement(this.renderer,this._el_0,'span',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_2_3 = new import14.Wrapper_NgClass(this.parent.parent.parent.parent.parentInjector.get(import16.IterableDiffers),this.parent.parent.parent.parent.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_2),this.renderer);
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n          ',(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_0,'click',this.eventHandler(this._handle_click_0_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.NgClass) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_2_3.context; }
    if (((token === import19.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = import4.interpolate(1,'btn btn-sm ',this.parent.context.$implicit.customClass,'');
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_8(((this.parent.parent.parent.parent.parent.context.isBS4 && !this.parent.context.$implicit.selected) && !this.parent.parent.parent.parent.parent.context.datePicker.isActive(this.parent.context.$implicit)),this.parent.context.$implicit.selected,this.parent.context.$implicit.disabled,(!this.parent.parent.parent.parent.parent.context.isBS4 && this.parent.parent.parent.parent.parent.context.datePicker.isActive(this.parent.context.$implicit)),!this.parent.parent.parent.parent.parent.context.isBS4);
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.detectChangesInInputProps(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this._map_9((this.parent.context.$implicit.secondary || this.parent.context.$implicit.current),(!this.parent.parent.parent.parent.parent.context.isBS4 && this.parent.context.$implicit.current));
    this._NgClass_2_3.check_ngClass(currVal_2_0_0,throwOnChange,false);
    this._NgClass_2_3.detectChangesInInputProps(this,this._el_2,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_7:any = this.parent.context.$implicit.disabled;
    if (import4.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this.renderer.setElementProperty(this._el_0,'disabled',currVal_7);
      this._expr_7 = currVal_7;
    }
    this._NgClass_0_3.detectChangesInHostProps(this,this._el_0,throwOnChange);
    this._NgClass_2_3.detectChangesInHostProps(this,this._el_2,throwOnChange);
    const currVal_10:any = import4.interpolate(1,'',this.parent.context.$implicit.label,'');
    if (import4.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this.renderer.setText(this._text_3,currVal_10);
      this._expr_10 = currVal_10;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_0_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0_0:any = ((<any>this.parent.parent.parent.parent.parent.context.datePicker.select(this.parent.context.$implicit.date)) !== false);
    return (true && pd_0_0);
  }
}
function viewFactory_DayPickerComponent8(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_DayPickerComponent8(viewUtils,parentInjector,declarationEl);
}
