// ng2-bootstrap/components/datepicker/yearpicker.component.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from 'ng2-bootstrap/components/datepicker/yearpicker.component';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/render/api';
import * as import3 from '@angular/core/src/linker/element';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ng2-bootstrap/components/datepicker/datepicker-inner.component';
import * as import9 from '@angular/core/src/metadata/view';
import * as import10 from '@angular/core/src/linker/component_factory';
import * as import11 from '../../../@angular/common/src/directives/ng_if.ngfactory';
import * as import12 from '@angular/core/src/linker/template_ref';
import * as import13 from '@angular/common/src/directives/ng_if';
import * as import14 from '../../../@angular/common/src/directives/ng_class.ngfactory';
import * as import15 from '../../../@angular/common/src/directives/ng_for.ngfactory';
import * as import16 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import17 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import18 from '@angular/core/src/linker/element_ref';
import * as import19 from '@angular/common/src/directives/ng_class';
import * as import20 from '@angular/common/src/directives/ng_for';
export class Wrapper_YearPickerComponent {
  context:import0.YearPickerComponent;
  changed:boolean;
  constructor(p0:any) {
    this.changed = false;
    this.context = new import0.YearPickerComponent(p0);
  }
  detectChangesInInputProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this.changed;
    this.changed = false;
    if (!throwOnChange) { if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); } }
    return changed;
  }
  detectChangesInHostProps(view:import1.AppView<any>,el:any,throwOnChange:boolean):void {
  }
}
var renderType_YearPickerComponent_Host:import2.RenderComponentType = (null as any);
class _View_YearPickerComponent_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import3.AppElement;
  _YearPickerComponent_0_4:Wrapper_YearPickerComponent;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_YearPickerComponent_Host0,renderType_YearPickerComponent_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.selectOrCreateRenderHostElement(this.renderer,'yearpicker',import4.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this._appEl_0 = new import3.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_YearPickerComponent0(this.viewUtils,this.injector(0),this._appEl_0);
    this._YearPickerComponent_0_4 = new Wrapper_YearPickerComponent(this.parentInjector.get(import8.DatePickerInnerComponent));
    this._appEl_0.initComponent(this._YearPickerComponent_0_4.context,([] as any[]),compView_0);
    compView_0.create(this._YearPickerComponent_0_4.context,this.projectableNodes,(null as any));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.YearPickerComponent) && (0 === requestNodeIndex))) { return this._YearPickerComponent_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._YearPickerComponent_0_4.detectChangesInInputProps(this,this._el_0,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this._YearPickerComponent_0_4.detectChangesInHostProps(this,this._el_0,throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_YearPickerComponent_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  if ((renderType_YearPickerComponent_Host === (null as any))) { (renderType_YearPickerComponent_Host = viewUtils.createRenderComponentType('',0,import9.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_YearPickerComponent_Host0(viewUtils,parentInjector,declarationEl);
}
export const YearPickerComponentNgFactory:import10.ComponentFactory<import0.YearPickerComponent> = new import10.ComponentFactory<import0.YearPickerComponent>('yearpicker',viewFactory_YearPickerComponent_Host0,import0.YearPickerComponent);
const styles_YearPickerComponent:any[] = ([] as any[]);
var renderType_YearPickerComponent:import2.RenderComponentType = (null as any);
class _View_YearPickerComponent0 extends import1.AppView<import0.YearPickerComponent> {
  _text_0:any;
  _anchor_1:any;
  /*private*/ _appEl_1:import3.AppElement;
  _TemplateRef_1_5:any;
  _NgIf_1_6:import11.Wrapper_NgIf;
  _text_2:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_YearPickerComponent0,renderType_YearPickerComponent,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._text_0 = this.renderer.createText(parentRenderNode,'\n',(null as any));
    this._anchor_1 = this.renderer.createTemplateAnchor(parentRenderNode,(null as any));
    this._appEl_1 = new import3.AppElement(1,(null as any),this,this._anchor_1);
    this._TemplateRef_1_5 = new import12.TemplateRef_(this._appEl_1,viewFactory_YearPickerComponent1);
    this._NgIf_1_6 = new import11.Wrapper_NgIf(this._appEl_1.vcRef,this._TemplateRef_1_5);
    this._text_2 = this.renderer.createText(parentRenderNode,'\n  ',(null as any));
    this.init(([] as any[]),[
      this._text_0,
      this._anchor_1,
      this._text_2
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.TemplateRef) && (1 === requestNodeIndex))) { return this._TemplateRef_1_5; }
    if (((token === import13.NgIf) && (1 === requestNodeIndex))) { return this._NgIf_1_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = (this.context.datePicker.datepickerMode === 'year');
    this._NgIf_1_6.check_ngIf(currVal_1_0_0,throwOnChange,false);
    this._NgIf_1_6.detectChangesInInputProps(this,this._anchor_1,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
export function viewFactory_YearPickerComponent0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<import0.YearPickerComponent> {
  if ((renderType_YearPickerComponent === (null as any))) { (renderType_YearPickerComponent = viewUtils.createRenderComponentType('',0,import9.ViewEncapsulation.None,styles_YearPickerComponent,{})); }
  return new _View_YearPickerComponent0(viewUtils,parentInjector,declarationEl);
}
class _View_YearPickerComponent1 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _el_4:any;
  _text_5:any;
  _el_6:any;
  _text_7:any;
  _el_8:any;
  _text_9:any;
  _el_10:any;
  _text_11:any;
  _text_12:any;
  _text_13:any;
  _el_14:any;
  _text_15:any;
  _el_16:any;
  _NgClass_16_3:import14.Wrapper_NgClass;
  _text_17:any;
  _el_18:any;
  _text_19:any;
  _text_20:any;
  _text_21:any;
  _text_22:any;
  _el_23:any;
  _text_24:any;
  _el_25:any;
  _text_26:any;
  _el_27:any;
  _text_28:any;
  _text_29:any;
  _text_30:any;
  _text_31:any;
  _text_32:any;
  _el_33:any;
  _text_34:any;
  _anchor_35:any;
  /*private*/ _appEl_35:import3.AppElement;
  _TemplateRef_35_5:any;
  _NgFor_35_6:import15.Wrapper_NgFor;
  _text_36:any;
  _text_37:any;
  /*private*/ _expr_42:any;
  /*private*/ _expr_43:any;
  /*private*/ _expr_44:any;
  _map_45:any;
  /*private*/ _expr_46:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_YearPickerComponent1,renderType_YearPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_42 = import7.UNINITIALIZED;
    this._expr_43 = import7.UNINITIALIZED;
    this._expr_44 = import7.UNINITIALIZED;
    this._map_45 = import4.pureProxy1((p0:any):{[key: string]:any} => {
      return {disabled: p0};
    });
    this._expr_46 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'table',new import4.InlineArray2(2,'role','grid'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n  ',(null as any));
    this._el_2 = import4.createRenderElement(this.renderer,this._el_0,'thead',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n    ',(null as any));
    this._el_4 = import4.createRenderElement(this.renderer,this._el_2,'tr',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_5 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_6 = import4.createRenderElement(this.renderer,this._el_4,'th',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_7 = this.renderer.createText(this._el_6,'\n        ',(null as any));
    this._el_8 = import4.createRenderElement(this.renderer,this._el_6,'button',new import4.InlineArray8(6,'class','btn btn-default btn-sm pull-left','tabindex','-1','type','button'),(null as any));
    this._text_9 = this.renderer.createText(this._el_8,'\n          ',(null as any));
    this._el_10 = import4.createRenderElement(this.renderer,this._el_8,'i',new import4.InlineArray2(2,'class','glyphicon glyphicon-chevron-left'),(null as any));
    this._text_11 = this.renderer.createText(this._el_8,'\n        ',(null as any));
    this._text_12 = this.renderer.createText(this._el_6,'\n      ',(null as any));
    this._text_13 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_14 = import4.createRenderElement(this.renderer,this._el_4,'th',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_15 = this.renderer.createText(this._el_14,'\n        ',(null as any));
    this._el_16 = import4.createRenderElement(this.renderer,this._el_14,'button',new import4.InlineArray16(10,'class','btn btn-default btn-sm','role','heading','style','width:100%;','tabindex','-1','type','button'),(null as any));
    this._NgClass_16_3 = new import14.Wrapper_NgClass(this.parentInjector.get(import16.IterableDiffers),this.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_16),this.renderer);
    this._text_17 = this.renderer.createText(this._el_16,'\n          ',(null as any));
    this._el_18 = import4.createRenderElement(this.renderer,this._el_16,'strong',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_19 = this.renderer.createText(this._el_18,'',(null as any));
    this._text_20 = this.renderer.createText(this._el_16,'\n        ',(null as any));
    this._text_21 = this.renderer.createText(this._el_14,'\n      ',(null as any));
    this._text_22 = this.renderer.createText(this._el_4,'\n      ',(null as any));
    this._el_23 = import4.createRenderElement(this.renderer,this._el_4,'th',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_24 = this.renderer.createText(this._el_23,'\n        ',(null as any));
    this._el_25 = import4.createRenderElement(this.renderer,this._el_23,'button',new import4.InlineArray8(6,'class','btn btn-default btn-sm pull-right','tabindex','-1','type','button'),(null as any));
    this._text_26 = this.renderer.createText(this._el_25,'\n          ',(null as any));
    this._el_27 = import4.createRenderElement(this.renderer,this._el_25,'i',new import4.InlineArray2(2,'class','glyphicon glyphicon-chevron-right'),(null as any));
    this._text_28 = this.renderer.createText(this._el_25,'\n        ',(null as any));
    this._text_29 = this.renderer.createText(this._el_23,'\n      ',(null as any));
    this._text_30 = this.renderer.createText(this._el_4,'\n    ',(null as any));
    this._text_31 = this.renderer.createText(this._el_2,'\n  ',(null as any));
    this._text_32 = this.renderer.createText(this._el_0,'\n  ',(null as any));
    this._el_33 = import4.createRenderElement(this.renderer,this._el_0,'tbody',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_34 = this.renderer.createText(this._el_33,'\n    ',(null as any));
    this._anchor_35 = this.renderer.createTemplateAnchor(this._el_33,(null as any));
    this._appEl_35 = new import3.AppElement(35,33,this,this._anchor_35);
    this._TemplateRef_35_5 = new import12.TemplateRef_(this._appEl_35,viewFactory_YearPickerComponent2);
    this._NgFor_35_6 = new import15.Wrapper_NgFor(this._appEl_35.vcRef,this._TemplateRef_35_5,this.parentInjector.get(import16.IterableDiffers),this.parent.ref);
    this._text_36 = this.renderer.createText(this._el_33,'\n  ',(null as any));
    this._text_37 = this.renderer.createText(this._el_0,'\n',(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_8,'click',this.eventHandler(this._handle_click_8_0.bind(this)));
    var disposable_1:Function = this.renderer.listen(this._el_16,'click',this.eventHandler(this._handle_click_16_0.bind(this)));
    var disposable_2:Function = this.renderer.listen(this._el_25,'click',this.eventHandler(this._handle_click_25_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._el_6,
      this._text_7,
      this._el_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._text_12,
      this._text_13,
      this._el_14,
      this._text_15,
      this._el_16,
      this._text_17,
      this._el_18,
      this._text_19,
      this._text_20,
      this._text_21,
      this._text_22,
      this._el_23,
      this._text_24,
      this._el_25,
      this._text_26,
      this._el_27,
      this._text_28,
      this._text_29,
      this._text_30,
      this._text_31,
      this._text_32,
      this._el_33,
      this._text_34,
      this._anchor_35,
      this._text_36,
      this._text_37
    ]
    ,[
      disposable_0,
      disposable_1,
      disposable_2
    ]
    ,([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.NgClass) && ((16 <= requestNodeIndex) && (requestNodeIndex <= 20)))) { return this._NgClass_16_3.context; }
    if (((token === import12.TemplateRef) && (35 === requestNodeIndex))) { return this._TemplateRef_35_5; }
    if (((token === import20.NgFor) && (35 === requestNodeIndex))) { return this._NgFor_35_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_16_0_0:any = 'btn btn-default btn-sm';
    this._NgClass_16_3.check_klass(currVal_16_0_0,throwOnChange,false);
    const currVal_16_0_1:any = this._map_45((this.parent.context.datePicker.datepickerMode === this.parent.context.datePicker.maxMode));
    this._NgClass_16_3.check_ngClass(currVal_16_0_1,throwOnChange,false);
    this._NgClass_16_3.detectChangesInInputProps(this,this._el_16,throwOnChange);
    const currVal_35_0_0:any = this.parent.context.rows;
    this._NgFor_35_6.check_ngForOf(currVal_35_0_0,throwOnChange,false);
    this._NgFor_35_6.detectChangesInInputProps(this,this._anchor_35,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_42:any = (((this.parent.context.datePicker.yearColLimit - 2) <= 0)? 1: (this.parent.context.datePicker.yearColLimit - 2));
    if (import4.checkBinding(throwOnChange,this._expr_42,currVal_42)) {
      this.renderer.setElementAttribute(this._el_14,'colspan',((currVal_42 == (null as any))? (null as any): currVal_42.toString()));
      this._expr_42 = currVal_42;
    }
    const currVal_43:any = (this.parent.context.datePicker.uniqueId + '-title');
    if (import4.checkBinding(throwOnChange,this._expr_43,currVal_43)) {
      this.renderer.setElementProperty(this._el_16,'id',currVal_43);
      this._expr_43 = currVal_43;
    }
    const currVal_44:any = (this.parent.context.datePicker.datepickerMode === this.parent.context.datePicker.maxMode);
    if (import4.checkBinding(throwOnChange,this._expr_44,currVal_44)) {
      this.renderer.setElementProperty(this._el_16,'disabled',currVal_44);
      this._expr_44 = currVal_44;
    }
    this._NgClass_16_3.detectChangesInHostProps(this,this._el_16,throwOnChange);
    const currVal_46:any = import4.interpolate(1,'',this.parent.context.title,'');
    if (import4.checkBinding(throwOnChange,this._expr_46,currVal_46)) {
      this.renderer.setText(this._text_19,currVal_46);
      this._expr_46 = currVal_46;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_8_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_8_0:any = ((<any>this.parent.context.datePicker.move((0 - 1))) !== false);
    return (true && pd_8_0);
  }
  private _handle_click_16_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_16_0:any = ((<any>this.parent.context.datePicker.toggleMode()) !== false);
    return (true && pd_16_0);
  }
  private _handle_click_25_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_25_0:any = ((<any>this.parent.context.datePicker.move(1)) !== false);
    return (true && pd_25_0);
  }
}
function viewFactory_YearPickerComponent1(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_YearPickerComponent1(viewUtils,parentInjector,declarationEl);
}
class _View_YearPickerComponent2 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _appEl_2:import3.AppElement;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import15.Wrapper_NgFor;
  _text_3:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_YearPickerComponent2,renderType_YearPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'tr',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._appEl_2 = new import3.AppElement(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import12.TemplateRef_(this._appEl_2,viewFactory_YearPickerComponent3);
    this._NgFor_2_6 = new import15.Wrapper_NgFor(this._appEl_2.vcRef,this._TemplateRef_2_5,this.parent.parentInjector.get(import16.IterableDiffers),this.parent.parent.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n    ',(null as any));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ,([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import20.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = this.context.$implicit;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.detectChangesInInputProps(this,this._anchor_2,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    this.detectViewChildrenChanges(throwOnChange);
  }
}
function viewFactory_YearPickerComponent2(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_YearPickerComponent2(viewUtils,parentInjector,declarationEl);
}
class _View_YearPickerComponent3 extends import1.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _NgClass_2_3:import14.Wrapper_NgClass;
  _text_3:any;
  _el_4:any;
  _NgClass_4_3:import14.Wrapper_NgClass;
  _text_5:any;
  _text_6:any;
  _text_7:any;
  /*private*/ _expr_10:any;
  _map_11:any;
  _map_12:any;
  /*private*/ _expr_13:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement) {
    super(_View_YearPickerComponent3,renderType_YearPickerComponent,import6.ViewType.EMBEDDED,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
    this._expr_10 = import7.UNINITIALIZED;
    this._map_11 = import4.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        'btn-link': p0,
        'btn-info': p1,
        disabled: p2,
        active: p3
      }
      ;
    });
    this._map_12 = import4.pureProxy2((p0:any,p1:any):{[key: string]:any} => {
      return {
        'text-success': p0,
        'text-info': p1
      }
      ;
    });
    this._expr_13 = import7.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import3.AppElement {
    this._el_0 = import4.createRenderElement(this.renderer,(null as any),'td',new import4.InlineArray4(4,'class','text-center','role','gridcell'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n        ',(null as any));
    this._el_2 = import4.createRenderElement(this.renderer,this._el_0,'button',new import4.InlineArray8(8,'class','btn btn-default','style','min-width:100%;','tabindex','-1','type','button'),(null as any));
    this._NgClass_2_3 = new import14.Wrapper_NgClass(this.parent.parent.parentInjector.get(import16.IterableDiffers),this.parent.parent.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_2),this.renderer);
    this._text_3 = this.renderer.createText(this._el_2,'\n          ',(null as any));
    this._el_4 = import4.createRenderElement(this.renderer,this._el_2,'span',import4.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_4_3 = new import14.Wrapper_NgClass(this.parent.parent.parentInjector.get(import16.IterableDiffers),this.parent.parent.parentInjector.get(import17.KeyValueDiffers),new import18.ElementRef(this._el_4),this.renderer);
    this._text_5 = this.renderer.createText(this._el_4,'',(null as any));
    this._text_6 = this.renderer.createText(this._el_2,'\n        ',(null as any));
    this._text_7 = this.renderer.createText(this._el_0,'\n      ',(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_2,'click',this.eventHandler(this._handle_click_2_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._el_4,
      this._text_5,
      this._text_6,
      this._text_7
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import19.NgClass) && ((4 <= requestNodeIndex) && (requestNodeIndex <= 5)))) { return this._NgClass_4_3.context; }
    if (((token === import19.NgClass) && ((2 <= requestNodeIndex) && (requestNodeIndex <= 6)))) { return this._NgClass_2_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_2_0_0:any = 'btn btn-default';
    this._NgClass_2_3.check_klass(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this._map_11(((this.parent.parent.parent.context.isBS4 && !this.context.$implicit.selected) && !this.parent.parent.parent.context.datePicker.isActive(this.context.$implicit)),(this.context.$implicit.selected || ((this.parent.parent.parent.context.isBS4 && !this.context.$implicit.selected) && this.parent.parent.parent.context.datePicker.isActive(this.context.$implicit))),this.context.$implicit.disabled,(!this.parent.parent.parent.context.isBS4 && this.parent.parent.parent.context.datePicker.isActive(this.context.$implicit)));
    this._NgClass_2_3.check_ngClass(currVal_2_0_1,throwOnChange,false);
    this._NgClass_2_3.detectChangesInInputProps(this,this._el_2,throwOnChange);
    const currVal_4_0_0:any = this._map_12((this.parent.parent.parent.context.isBS4 && this.context.$implicit.current),(!this.parent.parent.parent.context.isBS4 && this.context.$implicit.current));
    this._NgClass_4_3.check_ngClass(currVal_4_0_0,throwOnChange,false);
    this._NgClass_4_3.detectChangesInInputProps(this,this._el_4,throwOnChange);
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_10:any = this.context.$implicit.disabled;
    if (import4.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this.renderer.setElementProperty(this._el_2,'disabled',currVal_10);
      this._expr_10 = currVal_10;
    }
    this._NgClass_2_3.detectChangesInHostProps(this,this._el_2,throwOnChange);
    this._NgClass_4_3.detectChangesInHostProps(this,this._el_4,throwOnChange);
    const currVal_13:any = import4.interpolate(1,'',this.context.$implicit.label,'');
    if (import4.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this.renderer.setText(this._text_5,currVal_13);
      this._expr_13 = currVal_13;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  private _handle_click_2_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_2_0:any = ((<any>this.parent.parent.parent.context.datePicker.select(this.context.$implicit.date)) !== false);
    return (true && pd_2_0);
  }
}
function viewFactory_YearPickerComponent3(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import3.AppElement):import1.AppView<any> {
  return new _View_YearPickerComponent3(viewUtils,parentInjector,declarationEl);
}
